Fix creditcard validation

saevarom opened the pull request.

saevarom committed:
Fix creditcard validation

gpassarelli commented:
Hey @saevarom, could you please let me know in which scenario the validation didn't work so I can try to replicate the error and test the fix?

Thanks.

saevarom commented:
Yes, I was using the creditcard validator to limit valid cards to Visa and Mastercard. So I have an input like this:

`<input data-parsley-required="true" data-parsley-creditcard="visa,mastercard" type="tel" id="cc" className="form-control" placeholder="Creditcard number" />`

The line where I changed `return` to `break` was causing the credit card detection function to return a `null` (or `undefined`, can't remember which) instead of breaking out of the loop and continue to run the rest of the function. The `return` statement always stops the exectution of the function while the `break` statement just breaks out of the current loop.

The second thing where I changed `c` to `card[c]` was to return the actual value of the item, instead of just the index `c`. Because when you do `for (var c in cards)` the `c` variable is just the index to the current value instead of being the actual value.

Hope this clears it up.
- Sævar
